### PR TITLE
GSoc 2020 report: Nipun Garg

### PR DESCRIPTION
**Pull request by nipunG314**
This is the report for the GSoC 2020 project: New Vulkan Samples to Extend Vulkan API Coverage.

### CODE REVIEW

File: docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-nipungarg.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+
+PRs:
+[#169](https://github.com/google/vulkan_test_applications/pull/169):
+This PR creates a simple overlapping frames sample where we first draw a triangle to a temporary Vulkan Image, and then invert its color in separate render passes.
+
+[#184](https://github.com/google/vulkan_test_applications/pull/184):
+This PR modifies the overlapping_frames sample for different image each frame.
+
+### Vulkan Test Applications - Combining Graphics and Compute with Raymarching and Noise Texture
+The idea behind this sample was to combine the Compute and Graphics capabilities of Vulkan in order to visualize 3D Noise Textures. For the sample, we would procedurally generate a 3D Perlin Texture using the Compute queue and then use the resulting texture and a given surface level as inputs for a Raymarching Renderer.
```
Comment (hevrard) [inline docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-nipungarg.md:318]: The idea ... *is* to combine ...

File: docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-nipungarg.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+This was the final sample that was planned under the project. However, I could not get it to work satisfactorily under the current schedule. Currently, the setup for Graphics of the project exists. However, the shaders need to be re-written as they do not work as desired right now. The Compute side of the project is unimplemented till now.
```
Comment (hevrard) [inline docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-nipungarg.md:320]: Replace the second "However" by "Still, ..."

`... till now` -> `... at the time of writing this report`.

File: docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-nipungarg.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+
+[Current Work Link](https://github.com/nipunG314/vulkan_test_applications/tree/noise_visual_raymarcher)
+
+I would like to spend a couple weeks after the program to get this up and running.
```
Comment (hevrard) [inline docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-nipungarg.md:324]: `I would like to` -> `I plan to`

(Glad to read this! But first we should make sure the overlapping frame sample is properly merged into the main branch)

**Review by hevrard — CHANGES_REQUESTED** [review]
Good, a last series of fixes then we're good to go!

File: docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-nipungarg.md (hunk 3 above)
Comment (nipunG314) [reply docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-nipungarg.md:324]: Sure. There is a current PR pending for that. And I have the code for the next PR ready for that.

**Comment by nipunG314** [comment]
@hevrard I think I've addressed all the issues outlined above. Could you please take a look?

**Review by hevrard — APPROVED** [review]

**Pull request merged**